packages/bsky/src/subscription/repo.ts

```diff
@@ -10,74 +10,110 @@ import Database from '../db'
 import AppContext from '../context'
 import { Leader } from '../db/leader'
 import { subLogger } from '../logger'
+import { ConsecutiveList, LatestQueue, PartitionedQueue } from './util'
 
 const METHOD = ids.ComAtprotoSyncSubscribeAllRepos
 export const REPO_SUB_ID = 1000
 
 export class RepoSubscription {
   leader = new Leader(this.subLockId, this.ctx.db)
+  repoQueue = new PartitionedQueue()
+  cursorQueue = new LatestQueue()
+  consecutive = new ConsecutiveList<Message>()
   destroyed = false
+
   constructor(
     public ctx: AppContext,
     public service: string,
     public subLockId = REPO_SUB_ID,
   ) {}
 
   async run() {
-    const { db } = this.ctx
     while (!this.destroyed) {
       try {
         const { ran } = await this.leader.run(async ({ signal }) => {
           const sub = this.getSubscription({ signal })
           for await (const msg of sub) {
-            try {
-              const ops = await getOps(msg)
-              await db.transaction(async (tx) => {
-                await Promise.all([
-                  this.handleOps(tx, ops, msg.time),
-                  this.handleActor(tx, msg.repo, msg.time),
-                ])
-                await this.setState(tx, { cursor: msg.seq })
+            const item = this.consecutive.push(msg)
+            this.repoQueue
+              .add(msg.repo, () => this.handleMessage(msg))
+              .catch((err) => {
+                subLogger.error(
```

Comment:
making sure i understand, we just drop messages we can't process, right?

Reply:
Yep that's right, I'll add a comment about that with a little rationale.